core.go

```diff
@@ -110,32 +106,90 @@ func (c *core) Write(ent zapcore.Entry, fs []zapcore.Field) error {
 
 	// We may be crashing the program, so should flush any buffered events.
 	if ent.Level > zapcore.ErrorLevel {
-		c.client.Flush(c.flushTimeout)
+		return c.Sync()
 	}
+
 	return nil
 }
 
+func (c *core) createExceptions() []sentry.Exception {
+	errorsCount := len(c.errs)
+
+	if errorsCount == 0 {
+		return nil
+	}
+
+	processedErrors := make(map[error]struct{}, errorsCount)
+	exceptions := make([]sentry.Exception, 0, errorsCount)
+
+	for i := errorsCount - 1; i >= 0; i-- {
+		exceptions = c.addExceptionsFromError(exceptions, processedErrors, c.errs[i])
+	}
+
+	if !c.cfg.DisableStacktrace && exceptions[0].Stacktrace == nil {
+		stacktrace := sentry.NewStacktrace()
+		if stacktrace != nil {
+			stacktrace.Frames = filterFrames(stacktrace.Frames)
+			exceptions[0].Stacktrace = stacktrace
+		}
+	}
+
+	// Reverse the exceptions; the most recent error must be the last one
+	for i := len(exceptions)/2 - 1; i >= 0; i-- {
+		j := len(exceptions) - 1 - i
+		exceptions[i], exceptions[j] = exceptions[j], exceptions[i]
+	}
+
+	return exceptions
+}
+
+func (c *core) addExceptionsFromError(
+	exceptions []sentry.Exception,
+	processedErrors map[error]struct{},
```

Comment:
should "processedErrors " be passed from the outside?

Reply:
It is shared between invocations, so yes. Its used to de-duplicate the errors.